### PR TITLE
ref(discover) Add an error_rate function

### PR DESCRIPTION
**Pull request by evanh**
The error_rate function is getting used more and more by the product, and it is
currently calculated on the sentry side instead of in snuba. Move the logic into
snuba so that we can properly use the AST and avoid edge case errors where the
events table is selected instead of the transactions table.

### CODE REVIEW

File: snuba/util.py [hunk 1]
```diff
@@ -97,6 +98,11 @@ def function_expr(fn: str, args_expr: str = "") -> str:
                 apdex=apdex, user_col=escape_identifier(match.group(3)),
             )
         raise ValueError("Invalid format for impact()")
+    elif fn.startswith("error_rate("):
+        match = ERROR_RATE_FUNCTION_RE.match(fn)
+        if match:
+            return "countIf((transaction_status != 0 AND transaction_status != 2)) / count()"
+        raise ValueError("Invalid format for error_rate()")
```
Comment (fpacifici) [inline snuba/util.py:105]: Same here, please use SPAN_STATUS_NAME_TO_CODE

**Review by fpacifici — CHANGES_REQUESTED** [review]

File: snuba/query/processors/error_rate_processor.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    """
+    A percentage of transactions with a bad status. "Bad" status is defined as anything other than 0 (success) and 2 (unknown).
+    See here (https://github.com/getsentry/relay/blob/master/py/sentry_relay/consts.py) for the full list of errors.
+
+    divide(countIf(and(notEquals(transaction_status, 0), notEquals(transaction_status, 2))), count())
+    """
+
+    def process_query(self, query: Query, request_settings: RequestSettings) -> None:
+        def process_functions(exp: Expression) -> Expression:
+            if isinstance(exp, FunctionCall) and exp.function_name == "error_rate":
```
[the remark this answers is not in the document]
Comment (evanh) [reply snuba/query/processors/error_rate_processor.py:27]: Sorry what enum are you talking about? SPAN_STATUS_NAME_TO_CODE?

[the remark this answers is not in the document]
Comment (fpacifici) [reply snuba/query/processors/error_rate_processor.py:27]: Yes, sorry it is not an enum. but that one.

File: snuba/query/processors/error_rate_processor.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                assert len(exp.parameters) == 0
+
+                return div(
+                    countIf(
+                        FunctionCall(
+                            None,
+                            "and",
```
Comment (fpacifici) [inline snuba/query/processors/error_rate_processor.py:34]: you can use `binary_condition` also to build an AND.  And the function to use is declared in BooleanFunctions in the same place as ConditionFunctions

**Review by fpacifici — APPROVED** [review]

File: snuba/query/processors/error_rate_processor.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (evanh) [reply snuba/query/processors/error_rate_processor.py:27]: There's "unknown_error", which is code 2. And yes that is supposed to be considered a success. Or at least not a failure.

**Pull request merged**